packages/headless/src/features/facets/category-facet-set/category-facet-reducer-helpers.ts

```diff
@@ -0,0 +1,42 @@
+import {
+  CategoryFacetRequest,
+  CategoryFacetValueRequest,
+} from './interfaces/request';
+
+export function selectPath(request: CategoryFacetRequest, path: string[]) {
```

Comment:
Indeed, the reason for extracting was for testability. The `selectPath` function has five UTs, and it used in two reducer cases.

If it was internal to the reducer file, then the honest approach to testing the two cases would be to repeat all 5 UTs since, as you said, the implementation is internal and we do not know what is happening.

By extracting, it becomes possible to spy on the function. The reducer cases still have a test that asserts the output, but also that `selectPath` was called. With these two assertions, I think we can avoid repeating all the tests related to `selectPath`.

What do you think?